lib/ansible/module_utils/appliance/ibm/idg_common.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017 Tecnologías Gallo Rojo.
+# GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.urls import url_argument_spec
+from ansible.module_utils._text import to_native
+
+import json
+# import pdb
+
+# Seed the result
+result = dict(
+    failed=False,
+    changed=False,
+    name='default',
+    msg='The best is coming'
+)
+
+# Socket information
+idg_endpoint_spec = url_argument_spec()
+idg_endpoint_spec.update(
+    timeout=dict(type='int', default=10),  # The socket level timeout in seconds
+    server=dict(type='str', required=True),  # Remote IDG will be used.
+    server_port=dict(type='int', default=5554),  # Remote IDG port be used.
+    # Credentials
+    user=dict(type='str', required=True),
+    password=dict(type='str', required=True, no_log=True)
+)
+
+del idg_endpoint_spec["url_username"]
+del idg_endpoint_spec["url_password"]
+
+
+# Custom exception
+class IDGException(Exception):
+    pass
+
+
+class IDGUtils(object):
+    """ Class class with very useful things """
+
+    #############################
+    # Constants
+    #############################
+
+    IMMUTABLE_MESSAGE = 'The current state is consistent with the desired configuration.'
+    COMPLETED_MESSAGE = 'Completed.'
+    CHECK_MODE_MESSAGE = 'Change was only simulated, due to enabling verification mode.'
+    UNCONTROLLED_EXCEPTION = 'Unknown exception.'
+
+    # Connection agreements
+    BASIC_HEADERS = {"Content-Type": "application/json"}
+    ANSIBLE_VERSION = '2.7'
```

Comment:
Their should be a variable you can read in, rather than hardcoding this

Reply:
Hi John!

IBM DataPower® Gateway serves as the enterprise-grade, field-proven application integration for SOA and REST and security-rich API gateway.
This technology has maintained steady growth and is seen as an important component of IBM's cloud strategy.

[https://www.ibm.com/products/datapower-gateway](https://www.ibm.com/products/datapower-gateway)

I also do not feel very comfortable with the current route, but I did not find a position that I could reuse, maybe something like: gateway, api, ...?

Thanks to the corrections, we will work to ensure that the appropriate corrections are made. 

Thank you for your attention.